pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -86,65 +87,88 @@ type MasterConfig struct {
+	_, portString, err := net.SplitHostPort(masterConfig.ServingInfo.BindAddress)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	port, err := strconv.Atoi(portString)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	portRange, err := knet.ParsePortRange(options.KubernetesMasterConfig.ServicesNodePortRange)
+	portRange, err := knet.ParsePortRange(masterConfig.KubernetesMasterConfig.ServicesNodePortRange)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	// Defaults are tested in TestAPIServerDefaults
-	server := apiserveroptions.NewServerRunOptions()
+	server := kapiserveroptions.NewServerRunOptions()
 
 	// Adjust defaults
 	server.EventTTL = 2 * time.Hour
-	server.GenericServerRunOptions.AnonymousAuth = true
-	server.GenericServerRunOptions.ServiceClusterIPRange = net.IPNet(flagtypes.DefaultIPNet(options.KubernetesMasterConfig.ServicesSubnet))
-	server.GenericServerRunOptions.ServiceNodePortRange = *portRange
-	server.GenericServerRunOptions.EnableProfiling = true
-	server.GenericServerRunOptions.MasterCount = options.KubernetesMasterConfig.MasterCount
-
-	server.GenericServerRunOptions.SecurePort = port
-	server.GenericServerRunOptions.InsecurePort = 0
-	server.GenericServerRunOptions.TLSCertFile = options.ServingInfo.ServerCert.CertFile
-	server.GenericServerRunOptions.TLSPrivateKeyFile = options.ServingInfo.ServerCert.KeyFile
-	server.GenericServerRunOptions.ClientCAFile = options.ServingInfo.ClientCA
-
-	server.GenericServerRunOptions.MaxRequestsInFlight = options.ServingInfo.MaxRequestsInFlight
-	server.GenericServerRunOptions.MinRequestTimeout = options.ServingInfo.RequestTimeoutSeconds
-	for _, nc := range options.ServingInfo.NamedCertificates {
-		sniCert := config.NamedCertKey{
+	server.ServiceClusterIPRange = net.IPNet(flagtypes.DefaultIPNet(masterConfig.KubernetesMasterConfig.ServicesSubnet))
+	server.ServiceNodePortRange = *portRange
+	server.Features.EnableProfiling = true
+	server.MasterCount = masterConfig.KubernetesMasterConfig.MasterCount
+
+	server.SecureServing.ServingOptions.BindPort = port
+	server.SecureServing.ServerCert.CertKey.CertFile = masterConfig.ServingInfo.ServerCert.CertFile
+	server.SecureServing.ServerCert.CertKey.KeyFile = masterConfig.ServingInfo.ServerCert.KeyFile
+	server.InsecureServing.BindPort = 0
+
+	// disable anonymous authentication
+	// NOTE: this is only to get rid of the "AnonymousAuth is not allowed with the AllowAll authorizer"
+	// warning. We do not use the authenticator or authorizer created by this.
+	server.Authentication.Anonymous.Allow = false
+	server.Authentication.ClientCert = &apiserveroptions.ClientCertAuthenticationOptions{masterConfig.ServingInfo.ClientCA}
+	if masterConfig.AuthConfig.RequestHeader == nil {
+		server.Authentication.RequestHeader = &genericoptions.RequestHeaderAuthenticationOptions{}
+	} else {
+		server.Authentication.RequestHeader = &genericoptions.RequestHeaderAuthenticationOptions{
```

Comment:
Because of https://github.com/openshift/origin/pull/13653#discussion_r111135242 and the change https://github.com/sttts/origin/pull/77/commits/d880bf248868403745cc0a26b347aaf8ffa60959